### DISCUSSION TITLE
Generalized IR Repeater

### DISCUSSION THREAD

**Discussion by rebel7580** [arendst/Tasmota#22533] [Q&A] [answered]
I was looking to do an IR repeater from one IR blaster to another. I looked at the cookbook examples for IR and RF repeaters, but they were limited to specific fields and required pre-knowledge of the received format.

Based on those examples, I made this rule for a JSON format that I see for my most common IR device:
```
ON IrReceived#Protocol DO var1 %value% endon
ON IrReceived#Bits DO var2 %value% endon
ON IrReceived#Data DO var3 %value% endon
ON IrReceived#DataLSB DO var4 %value% endon
ON IrReceived#Repeat DO PUBLISH cmnd/irblaster/irsend {"Protocol":"%var1%","Bits":"%var2%","Data":"%var3%",:DataLSB":"%var4%","Repeat":%value%"} endon
```
This works fine, but is limited to that specifc JSON format.

I'm wondering if there is a way to generalize this by taking the entire received JSON and simply re-publishing it without knowing what its structure is. I wasn't able to find a way to do that.

If I have: `ON IrReceived DO PUBLISH cmnd/irblaster/irsend "%value%"` the rule triggers, but `%value%` is empty.

Is this possible and I just missed where that is explained, or is it not doable?

#18910, #11985 discuss this but don't have the specific answer.

**Comment by barbudor** [comment]
Not with rules
You should use an ESP32* variant and Berry which can process the whole JSON whatever the format

**Reply by sfromis** [reply, marked as answer]
Without moving to ESP32 for the IR device, it would also be feasible to use another device (including but not limited to Tasmota on ESP32) to subscribe to the existing MQTT payloads, and do the replication publish from there.

**Reply by rebel7580** [reply]
I have two irblasters based on the 8266, so the Berry approach is out. 
Right now, it looks like the main signals I'm likely to want to repeat are all similar w.r.t the JSON, so my Rule above will work. 

Thanks for the comments.
